File: src/main/java/com/sydnie/mvc/services/BookService.java
package com.sydnie.mvc.services;
import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import com.sydnie.mvc.models.Book;
import com.sydnie.mvc.repositories.BookRepository;

@Service 
public class BookService {
	 // adding the book repository as a dependency
    private final BookRepository bookRepository;
    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }
    // returns all the books
    public List<Book> allBooks() {
        return bookRepository.findAll();
    }
 // creates a book
    public Book createBook(Book b) {
        return bookRepository.save(b);
    }
    // retrieves a book
    public Book findBook(Long id) {
        Optional<Book> optionalBook = bookRepository.findById(id);
        if(optionalBook.isPresent()) {
            return optionalBook.get();
        } else {
            return null;
        }
    }
// update a book
//    public Book updateById(Long id, String title, String description, String language, Integer pages) {
//    	Book foundBook = findBook(id);
//    	foundBook.setTitle(title);
//    	foundBook.setDescription(description);
//    	foundBook.setLanguage(language);
//    	foundBook.setNumberOfPages(pages);
//    	
//    	return bookRepository.save(foundBook);
//    }
    
    public Book updateBook(Book book) {
    	return bookRepository.save(book);
    }
    
 // delete a book
    public void deleteBook(Long id) {
    	bookRepository.deleteById(id);
    	
    }
  }
    

    
    

File: src/services/BookService.js
import axios from 'axios'
const REST_API_URL = 'http://localhost:8080/api/books';
class BookService {
    getBooks() {
        return axios.get(REST_API_URL);
    }
    createBook(book) {
        return axios.post(REST_API_URL, book);
    }
    getById(id) {
        return axios.get(REST_API_URL + '/' + id);
    }
    updateBook(id, book) {
        return axios.put(REST_API_URL + '/' + id, book);
    }
    deleteBook(id) {
        return axios.delete(REST_API_URL + '/' + id);
    }

}
export default new BookService();
